### PR TITLE
feat: icon preview and sites input

### PR DESCRIPTION
**Pull request by Kodylow**
Fixes up meta manager to allow a better input for sites. Also removes the whole "required" aspect and just allows for cstom key values

## Summary by CodeRabbit

- **New Features**
	- Introduced `CustomMetaFields` and `SitesInput` components for enhanced metadata management.
	- Added icon validation functionality for custom fields.

- **Modifications**
	- Updated `IconPreview`, `MetaInput`, `MetaManager`, and `RequiredMeta` components for improved state management and rendering logic.
	- Added `setActiveTab` prop to `FederationTabsCard` and `ProposedMetas` components for better tab control.

- **Bug Fixes**
	- Enhanced validation logic across components to ensure accurate metadata handling.

- **Documentation**
	- Updated interfaces and component signatures for clarity and usability.

### CODE REVIEW

File: apps/router/src/guardian-ui/components/dashboard/tabs/meta/manager/CustomMetaFields.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      }
+      newMeta[newKey] = value;
+      return newMeta;
+    });
+
+    if (newKey === 'federation_icon_url') {
+      setIconValidity(true);
+      setLocalIconUrl(null);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (Kodylow) [reply apps/router/src/guardian-ui/components/dashboard/tabs/meta/manager/CustomMetaFields.tsx:59]: can add some catching, probably a minor edge case though

File: apps/router/src/guardian-ui/components/dashboard/tabs/meta/manager/RequiredMeta.tsx [hunk 2]
```diff
@@ -1,79 +1,71 @@
...
-    } else {
-      setIconPreview(null);
-      setIsIconValid(true);
+      setRequiredMeta({
+        federation_name,
+        welcome_message,
+        federation_icon_url,
+      });
     }
+  }, [consensusMeta]);
```
[the remark this answers is not in the document]
Comment (Kodylow) [reply apps/router/src/guardian-ui/components/dashboard/tabs/meta/manager/RequiredMeta.tsx:38]: it always is

File: apps/router/src/guardian-ui/components/dashboard/tabs/meta/manager/MetaManager.tsx [hunk 3]
```diff
@@ -38,110 +30,83 @@ export const MetaManager = React.memo(function MetaManager({
...
-        federation_name,
-        welcome_message,
-        popup_end_timestamp,
-        federation_icon_url,
-      });
-      setOptionalMeta(rest);
+      const { sites = '[]', ...rest } = metaObj;
+
+      setSites(sites);
+      setCustomMeta(rest);
```
[the remark this answers is not in the document]
Comment (Kodylow) [reply apps/router/src/guardian-ui/components/dashboard/tabs/meta/manager/MetaManager.tsx:42]: handled in sitesinput

File: apps/router/src/guardian-ui/components/dashboard/tabs/meta/manager/MetaManager.tsx [hunk 4]
```diff
@@ -38,110 +30,83 @@
...
   const proposeMetaEdits = useCallback(() => {
-    if (metaModuleId === undefined || isAnyRequiredFieldEmpty()) return;
+    if (metaModuleId === undefined || isAnyFieldEmpty()) return;
     const updatedMetaArray = Object.entries({
-      ...requiredMeta,
-      ...optionalMeta,
-    }).filter(([key, value]) => key !== 'popup_end_timestamp' || value !== '');
+      ...customMeta,
+      sites,
+    });
```
[the remark this answers is not in the document]
Comment (Kodylow) [reply apps/router/src/guardian-ui/components/dashboard/tabs/meta/manager/MetaManager.tsx:109]: already handled

**Review by dpc — APPROVED** [review]

**Pull request merged**